sycl/test-e2e/lit.cfg.py

```diff
@@ -39,12 +39,16 @@
 
 # test-mode: Set if tests should run normally or only build/run
 config.test_mode = lit_config.params.get("test-mode", "full")
+config.fallback_build_run_only = False
 if config.test_mode == "full":
     config.available_features.add("run-mode")
     config.available_features.add("build-and-run-mode")
 elif config.test_mode == "run-only":
     lit_config.note("run-only test mode enabled, only executing tests")
     config.available_features.add("run-mode")
+    if lit_config.params.get("build-instead-of-skip-run-only", False):
+        config.available_features.add("build-and-run-mode")
+        config.fallback_build_run_only = True
```

Comment:
could we name this something more descriptive? maybe `fallback_to_build_if_unsupported` or something?

Reply:
sounds good to me, thanks!